platforms/qcs9100-ride-sx/contents.in

```diff
@@ -0,0 +1,85 @@
+<?xml version="1.0" ?>
+<!--
+========================================================================
+  contents.in
+
+  General Description
+     Contains information about component builds for this target.
+     It will clone as contents.xml during build compilation process.
+
+  Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
+  SPDX-License-Identifier: BSD-3-Clause
+
+========================================================================
+-->
+
+<contents>
+  <product_flavors cmm_pf_var="PRODUCT_FLAVORS">
+    <pf>
+      <name>default</name>
+      <component>
+        <name>common</name>
+        <flavor>default</flavor>
+      </component>
+      <component>
+        <name>apps</name>
+        <flavor>default</flavor>
+      </component>
+    </pf>
+    <pf>
+      <name>sail_nor</name>
+      <component>
+        <name>common</name>
+        <flavor>sail_nor</flavor>
+      </component>
+    </pf>
+  </product_flavors>
+  <product_info>
+    <product_name>QCS9100.LE.1.0</product_name>
+    <chipid flavor="default" storage_type="ufs" flash_phase="1">QCS9100</chipid>
+    <chipid flavor="sail_nor" storage_type="spinor" flash_phase="2">QCS9100</chipid>
+    <additional_chipid>SA8775P,QCS9075</additional_chipid>
+    <meta_type>SPF</meta_type>
+  </product_info>
+  <builds_flat>
```

Comment:
After looking at both templates, do we need build_flat to be a part of it? It feels like all its contents can be autogenerated.

Reply:
right, we either should generate the elements directly from the Python tool, or perhaps even better, as suggest by @lool  earlier look at using some better templating like jinja2